### DISCUSSION TITLE
download nhost db to local env

### DISCUSSION THREAD

**Discussion by kratam** [nhost/nhost#875] [Q&A]
Use case: I want to replicate the nhost cloud db in my local dev env (not just the schema but the actual data).
Right now I have the same schema but outdated data in my dev env.

Problems I'm facing:
- I don't want to trigger hasura events or psql triggers (before/after insert/delete)
- I have circular foreign-key constraints so I have to disable psql constraints temporarily
- I want to avoid writing a custom script for every table/trigger because I have many and don't want to keep track of them

What would be the best approach?

**Comment by elitan** [comment]
This is an interesting use case. I've not tried this myself, but this is what I would try first:

1. Export Hasura metadata, then reset the metadata, so no Hasura Events are fired. Then once the data import is completed, you can import the metadata again.

![CleanShot 2022-08-08 at 08 31 55](https://user-images.githubusercontent.com/331818/183354020-c81a01ab-8271-4459-9b87-304417dbb206.png)


2. What's an example of such a circular FK constraint? Is it possible to re-model your schema, so you don't have such circular FK constraint? I think that would be your best bet as it would make inserting a data backup like this much more manageable.
3. See 2.

**Comment by kratam** [comment]
> What's an example of such a circular FK constraint?

I have a table called `rates`. Every rate can have a single parent rate, stored in column `parent_id`. This relationship is used in psql triggers, i.e. when a parent rate is updated, its child rates are updated as well.
